apps/s_server.c

```diff
@@ -1500,7 +1506,16 @@ int s_server_main(int argc, char *argv[])
         case OPT_KEYLOG_FILE:
             keylog_file = opt_arg();
             break;
-
+        case OPT_MAX_EARLY:
+            max_early_data = atoi(opt_arg());
+            if (max_early_data < 0) {
+                BIO_printf(bio_err, "Invalid value for max_early_data\n");
+                goto end;
+            }
+            break;
+        case OPT_EARLY_DATA:
```

Comment:
No, they do different and independent things. Each makes sense by itself. If you supply "-early_data" by itself then it just uses the default value for max early data. If you supply "-max_early_data" by itself then it will reject early data up to the specified number of max early data bytes (plus an allowance for the crypto overhead). If it receives more data than that then it aborts the handshake.